app/src/main/java/com/skyyo/samples/features/autoComplete/AutoCompleteViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package com.skyyo.samples.features.autoComplete
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.util.*
+import javax.inject.Inject
+
+private const val QUERY = "query"
+private const val SUGGESTIONS = "suggestions"
+private const val IS_EXPANDED = "isExpanded"
+
+@HiltViewModel
+class AutoCompleteViewModel @Inject constructor(
+    private val handle: SavedStateHandle
+) : ViewModel() {
+
+    val countries = provideCountries()
+    val query = handle.getStateFlow(QUERY, "")
+    val suggestions = handle.getStateFlow(SUGGESTIONS, countries)
+    val isExpanded = handle.getStateFlow(IS_EXPANDED, false)
+
+    fun onCountryEntered(input: String) {
+        handle[QUERY] = input
+
+        viewModelScope.launch(Dispatchers.Default) {
+            handle[SUGGESTIONS] = if (input.isEmpty()) {
+                countries.also { onExpandedChange(false) }
+            } else {
+                val filteredList = countries.filter { country ->
+                    country
+                        .lowercase()
+                        .startsWith(input.lowercase()) && country != input
+                }
+                filteredList.also { onExpandedChange(true) }
+            }
+        }
```

Comment:
viewModelScope.launch(Dispatchers.Default) { doWork }
viewModelScope.launch { withContext(Dispatchers.Default)  { doWork }
isn't it the same thing?

about questions above
1. Yes. and I can't do anything with it. check please 3 point with detail explanation 
2. done fixing 
3. No difference, I'll remove one of them
4. `native exposed dropdown menu `-> current implementation by jetpack compose which works very bad with large list of data, because it uses column instead of lazyColumn under hood. that's why you can see lags, which causes ANR, because I used large list of data. And i added this sample on screen especially to show that lagging compare to `AndroidView` and `custom implementation`
`custom exposed dropdown menu` -> my implementation. which based on implementation by jetpack compose. (so I just copy pasted most of code) The main work I did - replace column with lazyColumn which causes all the problems with lags.